pkg/mattermost/helpers.go

```diff
@@ -143,3 +148,25 @@ func setProbes(customLiveness, customStartup, customReadiness corev1.Probe) (*co
 
 	return liveness, startUp, readiness
 }
+
+func checkMattermostImageVersion(imageName, version string) error {
+	var re = regexp.MustCompile(`mattermost/mattermost-.*[team|enterprise]-edition$`)
+	if !re.MatchString(imageName) {
+		return errors.Errorf("not using the mattermost official images so cannot validate the version")
```

Comment:
Will this make it impossible to use the Operator with a custom-built image?

Reply:
I guess the `.*` in the middle of the pattern will accept anything so someone could always put a custom tag name in that part of the string.

Reply:
This is a good point. If someone wants to make their own custom mattermost image I would think we would want them to still be able to use our operator. Either way:
```suggestion
		return errors.Error("not using the mattermost official images so cannot validate the version")
```